sycl/unittests/queue/ShortcutFunctions.cpp

```diff
@@ -0,0 +1,235 @@
+                                        size_t size, void *ptr,
+                                        pi_uint32 num_events_in_wait_list,
+                                        const pi_event *event_wait_list,
+                                        pi_event *event) {
+  TestContext->BufferReadCalled = true;
+  return PI_SUCCESS;
+}
+
+pi_result redefinedEnqueueMemBufferCopy(pi_queue command_queue,
+                                        pi_mem src_buffer, pi_mem dst_buffer,
+                                        size_t src_offset, size_t dst_offset,
+                                        size_t size,
+                                        pi_uint32 num_events_in_wait_list,
+                                        const pi_event *event_wait_list,
+                                        pi_event *event) {
+  TestContext->BufferCopyCalled = true;
+  return PI_SUCCESS;
+}
+
+pi_result redefinedEnqueueMemBufferFill(pi_queue command_queue, pi_mem buffer,
+                                        const void *pattern,
+                                        size_t pattern_size, size_t offset,
+                                        size_t size,
+                                        pi_uint32 num_events_in_wait_list,
+                                        const pi_event *event_wait_list,
+                                        pi_event *event) {
+  TestContext->BufferFillCalled = true;
+  return PI_SUCCESS;
+}
+
+TEST(ShortcutFunctions, ShortcutsCallCorrectPIFunctions) {
+  unittest::PiMock Mock;
+  platform Plt = Mock.getPlatform();
+
+  Mock.redefine<detail::PiApiKind::piEnqueueMemBufferWrite>(
+      redefinedEnqueueMemBufferWrite);
+  Mock.redefine<detail::PiApiKind::piEnqueueMemBufferRead>(
+      redefinedEnqueueMemBufferRead);
+  Mock.redefine<detail::PiApiKind::piEnqueueMemBufferCopy>(
+      redefinedEnqueueMemBufferCopy);
+
+  Mock.redefine<detail::PiApiKind::piEnqueueMemBufferFill>(
+      redefinedEnqueueMemBufferFill);
+
+  context Ctx(Plt);
+  queue Q{Ctx, default_selector()};
+
+  constexpr std::size_t Size = 1;
+
+  // Queue.copy(accessor src, shared_ptr dest);
+  {
+    TestContext.reset(new TestCtx());
+
+    int Data[Size];
+    buffer<int> Buf(Data, Size);
+
+    accessor<int, 1, access::mode::read, access::target::device,
+             access::placeholder::true_t, ext::oneapi::accessor_property_list<>>
+        Src(Buf);
+    ASSERT_TRUE(Src.is_placeholder());
+
+    std::shared_ptr<int> Dest = std::make_shared<int>(0);
+
+    Q.copy(Src, Dest);
+    Q.wait();
+
+    EXPECT_TRUE(TestContext->BufferReadCalled);
+  }
+
+  // Queue.copy(shared_ptr src, accessor dest);
+  {
+    TestContext.reset(new TestCtx());
+
+    int Data[Size];
+    buffer<int> Buf(Data, Size);
+
+    std::shared_ptr<int> Src = std::make_shared<int>(42);
+
+    accessor<int, 1, access::mode::write, access::target::device,
+             access::placeholder::true_t, ext::oneapi::accessor_property_list<>>
```

Comment:
I think SYCL 2020 makes `access::placeholder::true_t` unnecessary. I'd suggest to either simplify or at least add as extra simple

```
  accessor Dest(Buf, sycl::write);
```

test.